### DISCUSSION TITLE
Performance regression in 4.1.0 due to gpSP --> mGBA

### DISCUSSION THREAD

**Discussion by plexecutor** [OnionUI/Onion#635] [General]
### Checklist

- [X] I am reporting a bug in the Onion OS (Note that Onion devs can’t fix inherited issues with RA Cores)
- [X] I have reviewed the [Onion Wiki](https://github.com/OnionUI/Onion/wiki) and did not find the answer
- [X] I have searched the existing (open and closed) issues to ensure I do not log a duplicate

### Onions OS Version

4.1.0-alpha-dev

### Miyoo Firmware Version

202204190828

### Provide a clear and concise description of the issue

My favorite game and the game I play by far the most of on my Miyoo Mini is Mario Golf: Advanced Tour. While the performance wasn't flawless on 4.0.3 with the gpSP core, testing the game under 4.1.0-alpha-dev (latest release) has shown major performance issues with the game. I am getting frame dips and music stutters almost constantly while golfing. I personally don't feel like mGBA is ready for primetime on the Miyoo as it simply can't perform as well as gpSP on the limited hardware we have.

### Steps to Reproduce

Load Mario Golf: Advanced Tour on Onion 4.1.0-alpha-dev and play a round of golf.

### Provide any additional context or information that may help us investigate your issue

As I said above, gpSP runs the game in a state that I'd consider playable. mGBA does not.

**Comment by schmurtzm** [comment]
Hi, mGBA will bring a better compatibility for many games which is very welcome improvement. 
But yes it's true that the performances are a less good on the most demanding games. Fortunately Onion 4.1 arrive with a new feature : GLO menu allows you to change the default core for the few games were the performances are not good. 
On the last 4.1, highlight the game in the Game List and press the Y button to choose Gpsp core by default for Mario Golf. It should do the trick for the few exceptions that do not run correctly on mGBA.
An alternative consist to activate frameskip auto, I've tried on mario golf, it works well. I would not recommend it on every games but it seems OK to use frameskip with a golf game.

**Reply by plexecutor** [reply]
I understand the change was made for better compatibility but it comes at the cost of performance in many popular games. I personally feel like the core that runs the most commonly played games the best should be the default in order to create the best experience out of the box. Then if someone has a compatibility issue with a more obscure game, they can use the new core select feature to remedy that. Just my two cents.

**Reply by emperordahc** [reply]
I moved over to mGBA months ago; I had too many games that didn't run at all in gpSP.

The new GLO menu is a really great fix for the few games that have speed issues in mGBA. Thanks!

**Reply by sectune** [reply]
I agree that the move to mGBA as default is a great idea. Many Pokémon romhacks I play are not running on gpSP, just another example of better comp. For select games the user can still choose gpSP for better performance (not running > performance any day imho)

**Comment by clort81** [comment]
Retroarch allows associating a core with a rom file.  Could OnionUI take advantage of this?

It would be most preferable for OnionUI _remember_ *which core should run which game*.

I also want to mention a GBA use-case that has not received attention in this thread:

mGBA, with the configuration overlay as included with Onion in end of 2024, on Miyoo Mini quite a poor choice for turn based RPGs.  These games benefit greatly from being able to fast forward quickly. With MGBa on, say, FFVI, i can only get a perhaps 50% speedup.  I can get more if I set frameskip to some  fixed high level.  
It would be nice to have a GBA setup for fast foward support for RPGs with their endless grind battles!  Maybe something can be done?
Cheers
